src/core/Table/cells/EditableCell.tsx

```diff
@@ -41,39 +41,59 @@ export type EditableCellProps<
 export const EditableCell = <T extends Record<string, unknown>>(
   props: EditableCellProps<T>,
 ) => {
-  const {
-    cellElementProps,
-    cellProps,
-    onCellEdit,
-    children,
-    onInput,
-    onBlur,
-    ...rest
-  } = props;
+  const { cellElementProps, cellProps, onCellEdit, children, ...rest } = props;
+
+  const sanitizeString = (text: string) => {
+    return text.replace(/(\r\n|\n|\r)+/gm, ' ');
+  };
+
+  const [value, setValue] = React.useState(() =>
+    sanitizeString(cellProps.value),
+  );
+  React.useEffect(() => {
+    setValue(sanitizeString(cellProps.value));
+  }, [cellProps.value]);
 
-  const [value, setValue] = React.useState(cellProps.value);
   const [isDirty, setIsDirty] = React.useState(false);
 
   return (
     <div
       {...cellElementProps}
       contentEditable
       suppressContentEditableWarning
+      {...rest}
       onInput={(e) => {
-        setValue(
-          // Removes new line symbols
-          (e.target as HTMLElement).innerText.replace(/(\r\n|\n|\r)/gm, ''),
-        );
+        setValue(sanitizeString((e.target as HTMLElement).innerText));
         setIsDirty(true);
-        onInput?.(e);
+        props.onInput?.(e);
       }}
       onBlur={(e) => {
         if (isDirty) {
           onCellEdit(cellProps.column.id, value, cellProps.row.original);
         }
-        onBlur?.(e);
+        props.onBlur?.(e);
+      }}
+      onKeyDown={(e) => {
+        // Prevents from adding HTML elements (div, br) inside a cell on Enter press
+        if (e.key === 'Enter') {
+          e.preventDefault();
+        }
+        props.onKeyDown?.(e);
+      }}
+      onPaste={(e) => {
+        e.preventDefault();
+        document.execCommand(
```

Comment:
![image](https://user-images.githubusercontent.com/81580355/133049361-4d184e32-314b-46c1-8964-ea7b017ae842.png)

Reply:
I know... but there isn't any alternatives in new APIs.